### DISCUSSION TITLE
Problems with app_commands

### DISCUSSION THREAD

**Discussion by Matieus** [Rapptz/discord.py#8020] [Q&A] [answered]
I've been trying to run app_commands for several days. I tried all the ways with gist github, although the code starts and works and does not return errors, slash commands do not sync on discord servers.
I have the latest version of discord.py 2.0.0a downloaded today (10V 2022)
(Python 3.9.4)

To be sure if it is not the fault of the server settings or Integration on the virtual machine, I ran on discord.py 1.7.3 and discord_py_slash_command-3.0.3 and there the commands are visible on the servers.
(Python 3.8)

It looks like discord API is not receiving logged commands from discord.py 2.0.0a

Maybe I'm wrong or I'm doing something wrong, but please verify that the same is happening to you.

**Comment by AbstractUmbra** [comment]
How are you defining the commands, and how are you syncing the defined commands?

**Reply by Matieus** [reply]
I use this exmaple code
https://gist.github.com/AbstractUmbra/a9c188797ae194e592efe05fa129c57f#file-free_function_commands-py


import asyncio

from discord.ext import commands
from discord import app_commands

# define Bot with **needed** parameters
bot = commands.Bot(command_prefix="some_prefix", intents=some_intents_definition)

# You can now use `@bot.tree.command()` as a decorator:
@bot.tree.command()
async def my_command(interaction: discord.Interaction) -> None:
  await interaction.response.send_message("Hello from my command!")
### NOTE: the above is a global command, see the `main()` func below:

# we can even use Groups
group = app_commands.Group(name="some-parent", description="description")

@group.command()
async def my_subcommand(interaction: discord.Interaction) -> None:
  await interaction.response.send_message("hello from the subcommand!")
  
bot.tree.add_command(group, guild=discord.Object(id=...))

async def main():
  async with bot:
    # do you setup stuff if you need it here, then:
    bot.tree.copy_global_to(guild=discord.Object(id=...))  # we copy the global commands we have to a guild, this is optional
    await bot.start(MY_TOKEN)

**Reply by LostLuma** [reply]
As mentioned in my answer and the last line of the example you linked you will need to sync the tree to Discord. In the case of using `commands.Bot` the tree is an attribute that is created for you and available as [`Bot.tree`](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.Bot.tree).

**Reply by Matieus** [reply]
ok, but sync requires application_id how to put them when using commands.Bot?

**Reply by LostLuma** [reply]
You can pass the `application_id` parameter to the constructor.

**Comment by LostLuma** [comment, marked as answer]
The library does not automatically sync application commands to Discord, this is up to you as a developer.
You will need to call the [sync method](https://discordpy.readthedocs.io/en/latest/interactions/api.html#discord.app_commands.CommandTree.sync) on the tree you added your commands to after first creating or editing them to make them show up in app.

This could for example be done with a developer-only text or slash command, or a special flag when starting your application.
